tests/test_plugins.py

```diff
@@ -416,19 +416,20 @@ def test_plugin2_with_text_report(self):
 
         self.start_import_stop(cov, "caller")
 
-        repout = io.StringIO()
-        total = cov.report(file=repout, include=["*.html"], omit=["uni*.html"], show_missing=True)
-        report = repout.getvalue().splitlines()
-        expected = [
-            'Name         Stmts   Miss Branch BrPart  Cover   Missing',
-            '--------------------------------------------------------',
-            'bar_4.html       4      2      0      0    50%   1, 4',
-            'foo_7.html       7      5      0      0    29%   1-3, 6-7',
-            '--------------------------------------------------------',
-            'TOTAL           11      7      0      0    36%',
-        ]
-        assert expected == report
-        assert math.isclose(total, 4 / 11 * 100)
+        cov.save()
+        cov.load()
+        report = self.get_report(cov, squeeze=False,
+            include=["*.html"], omit=["uni*.html"], show_missing=True)
+        # Name          Stmts   Miss Branch BrPart  Cover  Missing
+        # --------------------------------------------------------
+        # bar_4.html        4      2      0      0    50% 1, 4
+        # foo_7.html        7      5      0      0    29% 1-3, 6-7
+        # --------------------------------------------------------
+        # TOTAL            11      7      0      0    36%
+        assert report.split("\n")[2] == \
+            "bar_4.html        4      2      0      0    50% 1, 4"
+        total = int(report.split("\n")[-2].split(" ")[-1][:-1])
```

Comment:
Why change the way the total is accessed by the test?

Reply:
I'll remove most of them and test the few cases that are necessary.